config/initializers/job_execution.rb

```diff
@@ -1,25 +1,28 @@
 if !Rails.env.test? && !ENV['PRECOMPILE']
   if ENV['SERVER_MODE']
     Rails.application.config.after_initialize do
-      Job.running.each(&:stop!) if Job.table_exists?
       JobExecution.enabled = true
-    end
-  end
 
-  Signal.trap('SIGUSR1') do
-    if JobExecution.enabled
-      # Disable new job execution
-      JobExecution.enabled = false
+      if Job.table_exists?
+        Job.running.each(&:stop!)
```

Comment:
If a job is running after a restart, that means the execution thread is gone. So just cancel it.